### ISSUE TITLE
expo build:android hangs indefinitely on iOS bundle

### ISSUE THREAD

**Issue by jamiesellick93** [expo/expo-cli#278]
## Description

The command `expo build:android` hangs indefinitely on `Building iOS bundle`. No errors are generated, it simply sits there.

## Expected Behavior

Expect an apk to be generated, or an error to be reported.

## Observed Behavior

Project setup checks pass, bundler starts, iOS bundle never finishes building.

If there's an error message, please paste the _full terminal output and error message_ in this code block:

```
[14:31:33] Making sure project is set up correctly...
[14:31:35] Your project looks good!
[14:31:36] Checking if current build exists...

[14:31:36] No currently active or previous builds for this project.
[14:31:37] Unable to find an existing Expo CLI instance for this directory, starting a new one...
[14:31:39] Starting Metro Bundler on port 19001.
[14:31:41] Tunnel ready.
[14:31:41] Publishing to channel 'default'...
[14:31:43] Building iOS bundle
```

## Environment

```
  Expo CLI 2.6.14 environment info:
    System:
      OS: Windows 10
    Binaries:
      Yarn: 1.9.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.1.0.0 AI-173.4907809
```

## Reproducible Demo

This is occurring with a fresh app made using `expo init`, blank template.

**Comment by ssrballin** [comment]
Having this same issue. Has anyone found any solutions yet?

**Comment by spyshower** [comment]
Why does it even build an iOS bundle?

**Comment by AryanJ-NYC** [comment]
> Why does it even build an iOS bundle?

@spyshower That's explained here: https://forums.expo.io/t/exp-build-ios-creates-an-android-bundle/10565/2

**Comment by lehresman** [comment]
Any update on this?  The same thing happens for me on macOS with expo cli 2.19.5.  It just hangs forever building the iOS bundle.

```
[07:42:49] Checking if there is a build in progress...
[07:42:49] Unable to find an existing Expo CLI instance for this directory, starting a new one...
[07:42:51] Starting Metro Bundler on port 19001.
[07:42:52] Tunnel ready.
[07:42:52] Publishing to channel 'staging'...
[07:42:53] Building iOS bundle
```

I trued the NODE_OPTIONS suggestion above, but the problem persists.

**Comment by bravetarget** [comment]
having the same problem under same conditions as @lehresman ☝️ .. this started occurring after  updating to SDK v33 from v28

**Comment by alexyangjie** [comment]
This issue starts to happen 12 hours ago on the Azure DevOps server with latest turtle-cli. I guess it is the same issue. Any idea what might be the issue?

**Comment by hyochan** [comment]
Same here!

**Comment by jennya** [comment]
Same issue on Expo 33

**Comment by aiham** [comment]
I hit this issue a lot. I usually am able to get it building again by clearing the metro cache:

```
./node_modules/\@expo/xdl/binaries/osx/watchman/watchman watch-del-all
./node_modules/\@expo/xdl/binaries/osx/watchman/watchman shutdown-server
rm -rf $TMPDIR/metro-cache*
```

System info:

```$ yarn expo-cli diagnostics
yarn run v1.12.3
$ expo-cli diagnostics

  Expo CLI 2.21.2 environment info:
    System:
      OS: macOS 10.14.5
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.4.0 - /var/folders/6p/262jd1016qj1l7616c5d3mph0000gn/T/yarn--1562763739163-0.4935897290444853/node
      Yarn: 1.12.3 - /var/folders/6p/262jd1016qj1l7616c5d3mph0000gn/T/yarn--1562763739163-0.4935897290444853/yarn
      npm: 6.9.0 - ~/.nvm/versions/node/v12.4.0/bin/npm
    IDEs:
      Xcode: 10.2.1/10E1001 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^33.0.0 => 33.0.7
      react-native: https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz => 0.59.8```

**Comment by cdcv** [comment]
Same problem.
Workaround. I was able to resolve this and get the build to work by first running
expo start -c 
and then running
expo build:android -t app-bundle

**Comment by MaxXxiMast** [comment]
Not sure if this is the issue but i was having the same issue, just signing in to the expo client with the credentials through which i had previously published solved the issue for me.

**Comment by ramsgei** [comment]
I am getting the error when try to build android app. Why it's building ios app and throwing the error.
![Expo Issue](https://user-images.githubusercontent.com/39795346/76421575-e6cd1e00-63c9-11ea-9d64-af113b5fb95e.PNG)

**Comment by mkhaledche** [comment]
> I am getting the error when try to build android app. Why it's building ios app and throwing the error.
> ![Expo Issue](https://user-images.githubusercontent.com/39795346/76421575-e6cd1e00-63c9-11ea-9d64-af113b5fb95e.PNG)

I am having the same error as well. Can anyone help regarding this issue?

**Comment by brentvatne** [comment]
delete the .expo directory inside of your project

**Comment by mkhaledche** [comment]
> delete the .expo directory inside of your project

It worked. Thank you so much.

**Comment by brentvatne** [comment]
great. this bug occurs when the server is stopped but doesn't have a chance to clean itself up (eg: operating system crashes and hard restarts). it will be resolved by https://github.com/expo/expo-cli/pull/1678 soon

**Comment by dsokal** [comment]
> great. this bug occurs when the server is stopped but doesn't have a chance to clean itself up (eg: operating system crashes and hard restarts). it will be resolved by #1678 soon

#1678 has been merged. I'm closing this issue. If anyone experiences this bug again, we'll reopen it again.

**Comment by mslipenchuk267** [comment]
@dsokal 

Hey getting this error with 
- Expo SDK v38
- Expo CLI 3.24.1
- npm v12
- running action on macos-latest

My error is that this hangs indefinitely at 'Building iOS Bundle` exactly as mentioned above.

I tried the workaround by @cdcv but I get stuck on the `expo start -c` step and `expo build:ios` never executes for me.
Here is what my action looks like with @cdcv proposed fix.

<img width="542" alt="Screen Shot 2020-08-24 at 11 40 03 AM" src="https://user-images.githubusercontent.com/25811429/91065741-db2f8580-e5fe-11ea-9ce4-3d2a9ae9f184.png">

When I run this not as an action, but manually locally. I would start metro bundler in another terminal tab and then run 'expo build:ios`. This would fix my problem but I don't believe I have this option with github actions?
